src/hb-ot-layout-gsubgpos.hh

```diff
@@ -3340,6 +3340,18 @@ struct GSUBGPOS
     hb_set_subtract (lookup_indexes, &inactive_lookups);
   }
 
+  void collect_features (const hb_map_t *org_feature_index_map,
+                         hb_set_t       *new_feature_indexes /* OUT */) const
```

Comment:
I would update this method to also output the set of script and langsys indices which are retained (ie. a map from script index to a set of langsys indices). If that is placed into the subset plan then later one when you're subsetting Script and LanSys you don't need to redo the langsys comparison again to locate which ones to drop. This would also mean it's not necessary to ship around the org_feature_index_map with the subset plan.

If this method is changed to do that, then it would also make sense to rename this and the context, which will help distinguish it from the other collect_features. Suggested name:

prune_langsys